modules/waybar/hm.nix

```
@@ -47,51 +43,100 @@ in
     };
   };
 
-  config = lib.mkIf (config.stylix.enable && cfg.enable) {
-    programs.waybar.style =
-      with config.lib.stylix.colors.withHashtag;
-      ''
-        @define-color base00 ${base00}; @define-color base01 ${base01}; @define-color base02 ${base02}; @define-color base03 ${base03};
-        @define-color base04 ${base04}; @define-color base05 ${base05}; @define-color base06 ${base06}; @define-color base07 ${base07};
-
-        @define-color base08 ${base08}; @define-color base09 ${base09}; @define-color base0A ${base0A}; @define-color base0B ${base0B};
-        @define-color base0C ${base0C}; @define-color base0D ${base0D}; @define-color base0E ${base0E}; @define-color base0F ${base0F};
-
-        * {
-            font-family: "${config.stylix.fonts.${cfg.font}.name}";
-            font-size: ${builtins.toString config.stylix.fonts.sizes.desktop}pt;
-        }
-      ''
-      + lib.optionalString cfg.addCss (
-        ''
+  configElements = [
+    (
+      { cfg }:
+      {
+        programs.waybar.style = lib.mkIf (cfg.addCss && cfg.background != null) ''
           window#waybar, tooltip {
-              background: alpha(@base00, ${builtins.toString config.stylix.opacity.desktop});
-              color: @base05;
+              background: ${cfg.background};
           }
-
-          tooltip {
-              border-color: @base0D;
+        '';
+      }
+    )
+    (
+      {
+        cfg,
+        opacity,
+        _colors,
+      }:
+      {
+        stylix.targets.waybar.background = lib.mkIf cfg.addCss "alpha(@base00, ${builtins.toString opacity.desktop})";
```

```suggestion
        stylix.targets.waybar.background = "alpha(@base00, ${builtins.toString opacity.desktop})";
```

This doesn't need to be conditional - we already check `addCss` [here](https://github.com/nix-community/stylix/blob/29db376e0c11a8960578e22ee3749fc147213712/modules/waybar/hm.nix#L50) and the [other definition](https://github.com/nix-community/stylix/blob/29db376e0c11a8960578e22ee3749fc147213712/modules/waybar/hm.nix#L95) is not conditional

---

IMO it seems unnecessary